### PR TITLE
OrganizeExtensions refactor added

### PR DESCRIPTION
**Pull request by Anabra**
The OrganizeExtensions refactor currently supports only a limited number of extensions.
The whole package is thoroughly tested.

### CODE REVIEW

File: src/ast/Language/Haskell/Tools/AST/Instances/SemanticTraversal.hs [hunk 1]
```diff
@@ -1,5 +1,4 @@
 -- don't show warnings, Template Haskell generated code contains unused variables
-{-# OPTIONS_GHC -fno-warn-unused-matches #-}
```
Comment (Anabra) [inline src/ast/Language/Haskell/Tools/AST/Instances/SemanticTraversal.hs]: Probably backend issue.

File: src/ast/Language/Haskell/Tools/AST/Instances/SourceInfoTraversal.hs [hunk 2]
```diff
@@ -1,5 +1,4 @@
 -- don't show warnings, Template Haskell generated code contains unused variables
-{-# OPTIONS_GHC -fno-warn-unused-matches #-}
```
Comment (Anabra) [inline src/ast/Language/Haskell/Tools/AST/Instances/SourceInfoTraversal.hs]: Probably backend issue.

File: src/ast/Language/Haskell/Tools/AST/References.hs [hunk 3]
```diff
@@ -1,5 +1,4 @@
 -- don't show warnings, Template Haskell generated code contains unused variables
-{-# OPTIONS_GHC -fno-warn-unused-matches #-}
```
Comment (Anabra) [inline src/ast/Language/Haskell/Tools/AST/References.hs]: Probably backend issue.

File: src/daemon/Language/Haskell/Tools/Daemon/Update.hs [hunk 4]
```diff
@@ -1,12 +1,4 @@
-{-# OPTIONS_GHC -O0 #-} -- this is needed because of an optimization bug in GHC 8.2.1: https://ghc.haskell.org/trac/ghc/ticket/13413
```
Comment (Anabra) [inline src/daemon/Language/Haskell/Tools/Daemon/Update.hs]: Probably backend issue.

File: src/refactor/Language/Haskell/Tools/Refactor/Utils/Extensions.hs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{-# OPTIONS_GHC -fwarn-incomplete-patterns -Werror #-} -- all extensions should be matched
```
Comment (Anabra) [inline src/refactor/Language/Haskell/Tools/Refactor/Utils/Extensions.hs]: Probably backend issue.

File: src/daemon/Language/Haskell/Tools/Daemon/Update.hs (hunk 4 above)
Comment (nboldi) [reply src/daemon/Language/Haskell/Tools/Daemon/Update.hs]: Nope, representation is OK, but you explicitely remove the pragmas with this assignment `filePragmas & annListElems .= newPragmas`.

**Pull request closed without merging**